packages/nimble-components/src/table/specs/README.md

```diff
@@ -1,78 +1,267 @@
+
+-   Data API :
+    -   Define what the interface is for setting/getting data on the table, taking into account any framework-specific binding concerns
+    -   Illustrate how the data is hooked up to the TanStack API
+    -   How is data hierarchy provided and represented in the DOM
+-   Column Definitions
+    -   Define the interface we will provide for the column providers/components (i.e., width, sorting, allowSort, allowGrouping, etc...)
+    -   Define the base implementation for the column providers that other column providers can extend.
+    -   Define how we intend to support defining a column that uses information from multiple columns (e.g. a hyperlink column that uses data from one column for the URL and the data from another as the text to display)
+        -   What column gets used for sorting?
+        -   Does column API allow for user to supply custom sort function?
+    -   List the set of column providers that Nimble will provide and provide their respective APIs where unique (e.g., formatter for DateTime column)
+    -   Define the table-level APIs for setting the columns
+        -   Slot API on table?
+        -   Alternative/accompanying programmatic API?
+-   Headers
+    -   Define the anatomy of headers in the table DOM
+        -   Require specific component type (i.e. do we need to create a `nimble-table-header`)
+        -   What is the component to use for interaction? Outline Button? Ghost button?
+        -   Can clients customize header content?
+        -   What and where are the interactive mechanisms/indicators? Sort arrow, etc..
+-   Row Selection
+    -   Define the anatomy of row selection in the table DOM
+        -   Indeterminate checkbox at the far left of each row?
+        -   Selected row CSS/design
+    -   Define events raised when row selection changes/occurs
+    -   Define table-level row-selection API
 -   Grouping
--   Expand/collapse
--   Hierarchical data
+    -   Define interactive mechanism, if any, to provide grouping
+    -   Define table-level API for setting grouping
+    -   Define events raised when grouping changes
+    -   Describe how data hierarchy should work with grouping
+-   Sorting
+    -   Define the table-level API for setting sorting
+    -   Define events raised when sorting changes
+    -   Describe how sorting should with with hierarchical data
+-   Action Menu
+    -   Define how the action menu gets associated with a particular column
+    -   Define the table-level(column-level?) API(s) for applying an action menu to the table (slot, properties, etc...)
+
+_Properties_
+
+Placeholder
+
+_Events_
+
+Placeholder
+
+### Anatomy
+
+_Slots_
+
+-   `slottedColumns` (Placeholder for column slots)
+-   `actionMenu` (Placeholder for action menu)
+
+### Angular integration
+
+Angular support should be accomplished through the typical directive patterns.
+
+_Placeholder for other Angular considerations:_
```

Comment:
Do you expect additional Angular & Blazor considerations to be added here? If not, we should remove the placeholders. We can always add more content even if they aren't there, but I think it would be better to avoid so many placeholders if we aren't specifically calling the sections out as incomplete.

Reply:
I've removed them and added notes about what we should expect to do for the data API in Blazor.